### DISCUSSION TITLE
No tests match the provided pattern

### DISCUSSION THREAD

**Discussion by Ayoubioussoufa** [Cyfrin/foundry-full-course-cu#1568] [Q&A] [answered]
Hey All,

I am on lesson 7 testing the (TestFundFailsWithoutEnoughEth, TestFundUpdatesFundedDataStructure) and I am receiving an error saying "No tests match the provided pattern. However, I have defined my test in my test file FundMeTest.t.sol

```solidity
function TestFundFailsWithoutEnoughEth() public {
        vm.expectRevert();
        fundme.fund();
    }

    function TestFundUpdatesFundedDataStructure() public {
        fundme.fund{value: 10e18}();
        uint256 amountFunded = fundme.getAddressToAmountFunded(address(this));
        assertEq(amountFunded, 10e18);
    }
    ```
    
Command Line:

```
rhyzome@LAPTOP-Q7SKSFGE:/mnt/c/Users/Ayoub/Desktop/Foundry-23/foundry-fund-me-23$ forge test --mt TestFundFailsWithoutEnoughEth
[⠆] Compiling...
[⠒] Compiling 1 files with 0.8.19
[⠑] Solc 0.8.19 finished in 1.23s
Compiler run successful!

No tests match the provided pattern:
        match-test: `TestFundFailsWithoutEnoughEth
```
Same case with TestFundUpdatesFundedDataStructure.

Any thoughts or insights is much appreciated. Thank you in advance.

P.S: Previous tests work correctly

**Comment by EngrPips** [comment, marked as answer]
Hello @Ayoubioussoufa , i would make the test function start with a small letter instead of the capital letters you are using in the screenshot you shared. i mean you test name should look like something below 

```javascript
function testFundFailsWithoutEnoughEth() public {
        vm.expectRevert();
        fundme.fund();
    }

    function testFundUpdatesFundedDataStructure() public {
        fundme.fund{value: 10e18}();
        uint256 amountFunded = fundme.getAddressToAmountFunded(address(this));
        assertEq(amountFunded, 10e18);
    }
```
Rename them like i have done above and try re-run the test, I would be looking forward to hear back from you.

**Reply by Ayoubioussoufa** [reply]
Thank you so much @EngrPips , I stayed more than an hour trying to debug it but in vain :/.

**Reply by EngrPips** [reply]
Awesome, i am glad you get them to work now. keep crushing them !!!
